### PR TITLE
Add org.kde.neochat (NeoChat)

### PR DESCRIPTION
**Pull request by travier**
This is a WIP PR as there are currently no official release for NeoChat, but I have tried it and it is working as far as I have tested.
Ping @flathub/kde.

### CODE REVIEW

**Comment by travier** [comment]
bot, build org.kde.neochat

**Comment by tsdgeos** [comment]
I'd personally wait for an actual release, i think @ognarb is planning to do one soon-ish?

File: org.kde.neochat.json [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            "name": "extra-cmake-modules",
+            "buildsystem": "cmake-ninja",
+            "cleanup": [
+                "*"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/KDE/extra-cmake-modules/archive/v5.70.0.tar.gz",
```
Comment (tsdgeos) [inline org.kde.neochat.json:46]: ECM is part  of the sdk so you shouldn't need this

**Comment by travier** [comment]
> I'd personally wait for an actual release, i think @ognarb is planning to do one soon-ish?

Sure, I will update that one when it is released.

**Comment by TobiasFella** [comment]
FYI: We plan to release neochat 1.0 next weekend

File: org.kde.neochat.json [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        },
+        {
+            "name": "neochat",
+            "buildsystem": "cmake-ninja",
+            "builddir": true,
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://invent.kde.org/network/neochat.git",
+                    "branch": "master"
```
Comment (ognarb) [inline org.kde.neochat.json:90]: We now have a 1.0 stable branch, you can use it. The 1.0 tag will be created from this branch and the master branch now require libQuotient master.

Comment (travier) [reply org.kde.neochat.json:90]: Can we get a tag for kquickimageeditor too? Thanks!

**Comment by travier** [comment]
bot, build org.kde.neochat

**Comment by aleixpol** [comment]
While this is quite ideal, we won't be getting any translations if we build from branches/tags (unless neochat does like KF5 and includes translations in tagged snapshots).

We'll need to use the tarball.

**Comment by ognarb** [comment]
Tags are out and a release announcement too.

https://carlschwan.eu/2020/12/23/announcing-neochat-1.0-the-kde-matrix-client/

Additionally, you might want to add the following patch, because I forgot to add it in the package (boo me) :( 
https://invent.kde.org/network/neochat/commit/3fe10bfc3c202144804f0a81248d55f59fe4304a

**Comment by travier** [comment]
bot, build org.kde.neochat

**Comment by travier** [comment]
Updated for 1.0 release: https://carlschwan.eu/2020/12/23/announcing-neochat-1.0-the-kde-matrix-client/

**Comment by travier** [comment]
@ognarb Looks like something changed with the icon installation

**Comment by travier** [comment]
bot, build org.kde.neochat

**Comment by travier** [comment]
bot, build org.kde.neochat

**Comment by ognarb** [comment]
Thanks, it looks good from my side.

**Comment by travier** [comment]
Used a workaround for the icon issue. Somehow the PNG icons are not installed.
Latest push fixes desktop notification area support.

**Comment by ognarb** [comment]
> Used a workaround for the icon issue. Somehow the PNG icons are not installed.
> Latest push fixes desktop notification area support.

the pngs are for windows, the icon that should be installed is the org.kde.neochat.svg file.

**Comment by travier** [comment]
LGTM

**Comment by ognarb** [comment]
There was an issue during the package creation :( I will need to release kquickimageeditor 0.1.1 and neochat 1.0.1 :(

File: org.kde.neochat.json [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/kquickimageeditor/0.1/kquickimageeditor-0.1.tar.xz",
+                    "sha256": "ef000cd4d8006a1a0b204ea64ebed272b07d99e37734976d580580d87a31392b"
+                }
+            ]
+        },
+        {
+            "name": "neochat",
+            "buildsystem": "cmake-ninja",
+            "builddir": true,
```
Comment (ognarb) [inline org.kde.neochat.json:86]:
```suggestion
            "builddir": true,
            "config-opts": [
                "-DNEOCHAT_FLATPAK=ON"
            ],
```

**Comment by ognarb** [comment]
Ok the NeoChat package was fine but there was an error in the KQuickImageEditor package regarding the conversion (not a problem in Flatpak but it is one for distro package), so I needed to generate a new package 0.1.2.

And I almost forgot but the cmake flag -DNEOCHAT_FLATPAK=ON provides some special font config making the emojis render correctly in the Flatpak runtime.

**Comment by MayeulC** [comment]
> Build [35214 successful](https://flathub.org/builds/#/builders/32/builds/35214)
> To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/33900/org.kde.neochat.flatpakref
> ```

I've been trying this one, and see no statusbar icon, which might be due to missing d-bus paths?
https://invent.kde.org/network/neochat/-/issues/176

**Comment by barthalion** [comment]
Many apps use `--own-name=org.kde.*` due to semi-broken Qt tray support.

**Comment by ognarb** [comment]
Ok the new KQuickImageEditor version is: https://download.kde.org/stable/kquickimageeditor/0.1/kquickimageeditor-0.1.2.tar.xz. Sorry for the trouble, first releases are always more stressful than they should :/ 

These are the flatpak finish-args we are using in the upstream flatpak version: https://invent.kde.org/packaging/flatpak-kde-applications/-/blob/master/org.kde.neochat.json#L10

**Comment by barthalion** [comment]
> `--own-name=org.kde.StatusNotifierItem-2-2`

That `2-2` is the conflicting part, this is why you need `org.kde.*` for reliable tray support. It's major D-Bus hole but we couldn't figure out a better way.

**Comment by travier** [comment]
bot, build org.kde.neochat

**Comment by barthalion** [comment]
It seems to miss some icons for me:

image

At least the space left to the search bar seems to be clickable.

**Comment by MayeulC** [comment]
> It seems to miss some icons for me:
> 
> At least the space left to the search bar seems to be clickable.

This icon issue was linked to the theme issue for me: https://invent.kde.org/network/neochat/-/issues/177

Fixed (somewhat) in https://invent.kde.org/network/neochat/-/merge_requests/124

You could confirm it by trying to launch with `QT_QUICK_CONTROLS_STYLE=org.kde.desktop`.

**Comment by barthalion** [comment]
Yup, that fixes the issue for me.

**Comment by travier** [comment]
bot, build org.kde.neochat

**Comment by travier** [comment]
LGTM!

**Comment by travier** [comment]
Thanks!

**Comment by travier** [comment]
Can you make sure the @flathub/kde team has access? Thanks!

**Comment by barthalion** [comment]
Yeah, I already added the new repo to that team.

**Comment by travier** [comment]
From https://github.com/orgs/flathub/teams/kde/repositories?query=neo it looks like the team only has read access.

**Comment by barthalion** [comment]
Oops, fixed.

**Comment by travier** [comment]
Thanks!

**Pull request closed without merging**